pkg/types/types.go

```diff
@@ -201,23 +230,29 @@ type HostDevice interface {
 	GetDeviceCode() string
 }
 
-// PciDevice provides an interface to get PCI device specific information
+// PciDevice provides an interface to get generic PCI device information
 // extends HostDevice interface
 type PciDevice interface {
 	HostDevice
 	GetPciAddr() string
+}
+
+// NetDevice provides an interface to get generic network device information
+type NetDevice interface {
+	HostDevice
+	GetPFName() string
 	GetPfPciAddr() string
```

Comment:
can you move `GetPfPciAddr` and `GetVFID` to `PciDevice` ?

we can say that if a PciDevice does not have a parent PCI Address or a VF ID it will return empty string or -1 respectively WDYT ?


Also with all this refactoring and addition of interface i think we would want to add docstring to all methods so its clear what each method does.

Reply:
`AuxNetDevice` from AuxDevs PR use it, but `AccelDevice` don't. Keep it at `NetDevice` or move to `HostDevice`?

Reply:
OK, looking at aux dev PR i see that GetVFID is renamed to a more generic GetFuncID.
And GetPfPciAddr() is actually only used in the context of a Network device selectors.

lets keep as is.


> Also with all this refactoring and addition of interface i think we would want to add docstring to all methods so its clear what each method does.

lets just add docstrings then